android/src/com/appc/titanium/socketio/SocketIOClientProxy.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * This file was auto-generated by the Titanium Module SDK helper for Android
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2017 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ *
+ */
+package com.appc.titanium.socketio;
+
+import org.appcelerator.kroll.KrollDict;
+import org.appcelerator.kroll.KrollFunction;
+import org.appcelerator.kroll.KrollProxy;
+import org.appcelerator.kroll.annotations.Kroll;
+import org.appcelerator.kroll.common.Log;
+import org.appcelerator.kroll.common.TiConfig;
+import org.appcelerator.titanium.TiC;
+import org.appcelerator.titanium.util.TiConvert;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.WeakHashMap;
+
+import io.socket.client.Ack;
+import io.socket.client.Socket;
+import io.socket.client.SocketIOException;
+import io.socket.emitter.Emitter.Listener;
+
+@Kroll.proxy(creatableInModule=TiSocketioModule.class)
+public class SocketIOClientProxy extends KrollProxy
+{
+	// Standard Debugging variables
+	private static final String LCAT = "SocketIOClientProxy";
+	private static final boolean DBG = TiConfig.LOGD;
+
+	private Socket socket;
+	private WeakHashMap<KrollFunction, WeakReference<Listener>> callbackListenerMap;
+
+	// Constructor
+	public SocketIOClientProxy(Socket socket)
+	{
+		super();
+
+		this.socket = socket;
+		this.callbackListenerMap = new WeakHashMap<>();
+	}
+
+	@Kroll.getProperty
+	public boolean getConnected()
```

Comment:
`@Kroll.method` as well, to access it via `getConnected()` ?

Reply:
IMHO there is no benefit for an additional getter so i just stayed with the property access to `connected`.